index.js

```diff
@@ -12,17 +13,81 @@ const xpiZipFilename = stringToBytes('META-INF/mozilla.rsa');
 const oxmlContentTypes = stringToBytes('[Content_Types].xml');
 const oxmlRels = stringToBytes('_rels/.rels');
 
-const fileType = input => {
+/**
+ * Determine file type from Node.js stream
+ * @param {Readable} stream
+ * @returns {Promise<{ext, mime}|undefined>}
+ * @public
+ */
+async function fromStream(stream) {
+	const tokenizer = await strtok3.fromStream(stream);
+	try {
+		return await fromTokenizer(tokenizer);
+	} finally {
+		await tokenizer.close();
+	}
+}
+
+/**
+ * Determine file type from Buffer or Uint8Array.
+ * @param {Buffer | Uint8Array | ArrayBuffer} input - Partial file.
+ * @returns {Promise<{ext, mime}|undefined>} Promise with file-type
+ * @public
+ */
```

Comment:
I would prefer keeping the docs of the public methods to the TypeScript definition files.

Reply:
Sure, so no method comments. I can also convert to TypeScript if you prefer,